Makefile

```
@@ -62,6 +65,15 @@ setup: my.env .docker-build  ## | Set up services.
 run: my.env .docker-build  ## | Run the webapp and services.
 	${DC} up web fakesentry
 
+.PHONY: devcontainerbuild
+devcontainerbuild: .env .docker-build .devcontainer-build  ## | Build VS Code development container.
```

I don't think this should depend on `.devcontainer-build`, since that would be a cyclic dependency. I know we have the same for Tecken and it works, but it does look wrong, doesn't it?

---

Thinking about it, I don't see any reason why we would need the `.devcontainer-build` and `devcontainerbuild` targets at all. docker-compose will build the container when needed, and we can simply let it do its thing.

---

> docker-compose will build the container when needed

when you want to pick up changes to the devcontainer's Dockerfile you have to manually rebuild, or manually rm the existing image, because docker compose will continue using the existing image even if it's out of date

---

i got the dependencies wrong when copying from `make build`, it should be:
```suggestion
devcontainerbuild: .env  ## | Build VS Code development container.
```

---

Thanks, now I understand the use case. So far I've always used the "Rebuild Container" action inside VS Code to rebuild the container if needed, but I can see that it may be convenient to do this from the command line.

---

at least for me on mac that action seems to rebuild the *container* while re-using the image, but maybe it's supposed to do both and it's just that vscode has never built the image for me, because devcontainer doesn't work at all until i manually build the image.